Public/Src/Engine/UnitTests/Scheduler/PipFingerprinterTests.cs

```diff
@@ -400,6 +401,94 @@ public void TestUserProfileStaysOutOfFingerprints()
             XAssert.IsTrue(json.Contains(Path.GetFileName(Environment.GetFolderPath(Environment.SpecialFolder.InternetCache)).ToUpperInvariant()));
         }
 
+        private ContentFingerprint GetWeakFingerPrintForExecutable(
+            string exectuablePath, 
+            bool assertValue,  // true if fingerprint should contain executable path, false if output should be untracked

```

Comment:
I suggest removing this argument and removing all assertions (i.e., calls to `XAssert` methods) from this method.

The name `GetWeakFingerPrintForExecutable` suggests that this method is pure, i.e., that it will just compute the fingerprint, return it, and not have any other side effects.  

That said, I don't think it's necessary to check if the fingerprint text contains the executable path or not.  The real functional test specification is what you have below, i.e., asserting that (1) when the executable changes and it is not untracked => the fingerprints change, and (2) when the executable changes and it is untracked => the fingerprints don't change.